packages/grid/_modules_/grid/hooks/root/useContainerProps.ts

```diff
@@ -138,13 +138,18 @@ export const useContainerProps = (windowRef: React.RefObject<HTMLDivElement>, ap
 
   const updateContainerState = React.useCallback(
     (containerState: ContainerProps | null) => {
+      let updated = false;
       setGridState((state) => {
         if (!isEqual(state.containerSizes, containerState)) {
           state.containerSizes = containerState;
-          forceUpdate();
+          updated = true;
+          return { ...state };
```

Comment:
If we do want to treat the state immutable, better not mutate the old one (`.containerSizes`).